Compile time warnings

If i remember correctly casacore has always been quite tidy with compile time warnings, but if I compile the latest master I encounter a lot of warnings. We should probably solve these before the 2.5 release and maybe let travis fail on compile warnings.

```
/home/gijs/Work/casacore/casa/Containers/RecordDescRep.cc: In member function ‘casacore::String casacore::RecordDescRep::uniqueName(const casacore::String&) const’:      
/home/gijs/Work/casacore/casa/Containers/RecordDescRep.cc:374:8: warning: ‘%i’ directive writing between 1 and 10 bytes into a region of size 7 [-Wformat-overflow=]      
 String RecordDescRep::uniqueName (const String& name) const                                                                                                              
        ^~~~~~~~~~~~~                                                                                                                                                     
/home/gijs/Work/casacore/casa/Containers/RecordDescRep.cc:374:8: note: directive argument in the range [1, 2147483647]                                                    
In file included from /usr/include/stdio.h:862:0,                                                                                                                         
                 from /usr/include/c++/7/cstdio:42,                                                                                                                       
                 from /usr/include/c++/7/ext/string_conversions.h:43,                                                                                                     
                 from /usr/include/c++/7/bits/basic_string.h:6352,                                                                                                        
                 from /usr/include/c++/7/string:52,                                                                                                                       
                 from /usr/include/c++/7/bits/locale_classes.h:40,                                                                                                        
                 from /usr/include/c++/7/bits/ios_base.h:41,                                                                                                              
                 from /usr/include/c++/7/ios:42,                                                                                                                          
                 from /usr/include/c++/7/istream:38,                                                                                                                      
                 from /usr/include/c++/7/sstream:38,                                                                                                                      
                 from /usr/include/c++/7/complex:45,                                                                                                                      
                 from /home/gijs/Work/casacore/casacore/casa/BasicSL/Complexfwd.h:45,                                                                                     
                 from /home/gijs/Work/casacore/casacore/casa/BasicSL/Complex.h:35,                                                                                        
                 from /home/gijs/Work/casacore/casacore/casa/Utilities/DataType.h:32,                                                                                     
                 from /home/gijs/Work/casacore/casacore/casa/Containers/RecordDescRep.h:35,                                                                               
                 from /home/gijs/Work/casacore/casa/Containers/RecordDescRep.cc:28:                                                                                       
/usr/include/x86_64-linux-gnu/bits/stdio2.h:34:43: note: ‘__builtin___sprintf_chk’ output between 3 and 12 bytes into a destination of size 8                             
       __bos (__s), __fmt, __va_arg_pack ());                                                                                                                             
                                           ^                                                                                                                              
```

```
In member function ‘virtual void casacore::MultiFile::extend(casacore::MultiFileInfo&, casacore::Int64)’:                                                                              
cc1plus: warning: ‘void* __builtin_memset(void*, int, long unsigned int)’: specified size 18446744073709551608 exceeds maximum object size 9223372036854775807 [-Wstringop-overflow=]  
```

```
/home/gijs/Work/casacore/mirlib/key.c:439:9: warning: ignoring return value of ‘system’, declared with attribute warn_unused_result [-Wunused-result]    
         (void)system(string);                                                                                                                           
         ^~~~~~~~~~~~~~~~~~~~                                                                                                                            
/home/gijs/Work/casacore/mirlib/key.c:443:9: warning: ignoring return value of ‘system’, declared with attribute warn_unused_result [-Wunused-result]    
         (void)system(string);                                                                                                                           
         ^~~~~~~~~~~~~~~~~~~~                                                                                                                            
```

```
/home/gijs/Work/casacore/mirlib/uvio.c: In function ‘uvopen_c’:                                                                                           
/home/gijs/Work/casacore/mirlib/uvio.c:749:29: warning: passing argument 3 of ‘rdhdl_c’ from incompatible pointer type [-Wincompatible-pointer-types]     
       rdhdl_c(*tno,"vislen",&(uv->max_offset),hsize_c(uv->item));                                                                                        
                             ^                                                                                                                            
In file included from /home/gijs/Work/casacore/mirlib/uvio.c:288:0:                                                                                       
/home/gijs/Work/casacore/mirlib/miriad.h:182:6: note: expected ‘int8 * {aka long long int *}’ but argument is of type ‘off_t * {aka long int *}’          
 void rdhdl_c (int tno, Const char *keyword, int8 *value, int8 defval);                                                                                   
      ^~~~~~~                                                                                                                                             
/home/gijs/Work/casacore/mirlib/uvio.c:794:29: warning: passing argument 3 of ‘rdhdl_c’ from incompatible pointer type [-Wincompatible-pointer-types]     
       rdhdl_c(*tno,"vislen",&(uv->offset),hsize_c(uv->item));                                                                                            
                             ^                                                                                                                            
In file included from /home/gijs/Work/casacore/mirlib/uvio.c:288:0:                                                                                       
/home/gijs/Work/casacore/mirlib/miriad.h:182:6: note: expected ‘int8 * {aka long long int *}’ but argument is of type ‘off_t * {aka long int *}’          
 void rdhdl_c (int tno, Const char *keyword, int8 *value, int8 defval);                                                                                   
      ^~~~~~~                                                                                                                                             
/home/gijs/Work/casacore/mirlib/uvio.c:814:26: warning: passing argument 3 of ‘rdhdl_c’ from incompatible pointer type [-Wincompatible-pointer-types]     
     rdhdl_c(*tno,"ncorr",&(uv->corr_flags.offset),-1);                                                                                                   
                          ^                                                                                                                               
In file included from /home/gijs/Work/casacore/mirlib/uvio.c:288:0:                                                                                       
/home/gijs/Work/casacore/mirlib/miriad.h:182:6: note: expected ‘int8 * {aka long long int *}’ but argument is of type ‘off_t * {aka long int *}’          
 void rdhdl_c (int tno, Const char *keyword, int8 *value, int8 defval);                                                                                   
      ^~~~~~~                                                                                                                                             
/home/gijs/Work/casacore/mirlib/uvio.c:815:27: warning: passing argument 3 of ‘rdhdl_c’ from incompatible pointer type [-Wincompatible-pointer-types]     
     rdhdl_c(*tno,"nwcorr",&(uv->wcorr_flags.offset),-1);                                                                                                 
                           ^                                                                                                                              
In file included from /home/gijs/Work/casacore/mirlib/uvio.c:288:0:                                                                                       
/home/gijs/Work/casacore/mirlib/miriad.h:182:6: note: expected ‘int8 * {aka long long int *}’ but argument is of type ‘off_t * {aka long int *}’          
 void rdhdl_c (int tno, Const char *keyword, int8 *value, int8 defval);                                                                                   
      ^~~~~~~                                                                                                                                             
```

```
/home/gijs/Work/casacore/tables/Tables/ReadAsciiTable.cc: In static member function ‘static void casacore::ReadAsciiTable::getTypes(const casacore::IPosition&, const Char*, casacore::Int, casacore::Char*, casacore::Char*│
, casacore::Char)’:                                                                                                                                                                                                         │
/home/gijs/Work/casacore/tables/Tables/ReadAsciiTable.cc:164:6: warning: ‘%i’ directive writing between 1 and 10 bytes into a region of size 9 [-Wformat-overflow=]                                                         ├
 void ReadAsciiTable::getTypes (const IPosition& shape,                                                                                                                                                                     │
      ^~~~~~~~~~~~~~                                                                                                                                                                                                        │
/home/gijs/Work/casacore/tables/Tables/ReadAsciiTable.cc:164:6: note: directive argument in the range [1, 2147483647]                                                                                                       
In file included from /usr/include/stdio.h:862:0,                                                                                                                                                                           │
                 from /usr/include/c++/7/cstdio:42,                                                                                                                                                                         │
                 from /usr/include/c++/7/ext/string_conversions.h:43,                                                                                                                                                       │
                 from /usr/include/c++/7/bits/basic_string.h:6352,                                                                                                                                                          │
                 from /usr/include/c++/7/string:52,                                                                                                                                                                         │
                 from /home/gijs/Work/casacore/casacore/casa/BasicSL/String.h:35,                                                                                                                                           │
                 from /home/gijs/Work/casacore/casacore/tables/Tables/ReadAsciiTable.h:33,                                                                                                                                  │
                 from /home/gijs/Work/casacore/tables/Tables/ReadAsciiTable.cc:28:                                                                                                                                          
/usr/include/x86_64-linux-gnu/bits/stdio2.h:34:43: note: ‘__builtin___sprintf_chk’ output between 9 and 18 bytes into a destination of size 16                                                                              │
       __bos (__s), __fmt, __va_arg_pack ());                                                                                                                                                                               
                                           ^                                                                                                                                                                                
```

```
/home/gijs/Work/casacore/tables/TaQL/test/tRecordGram.cc: In function ‘void testExpr2()’:                                                                
/home/gijs/Work/casacore/tables/TaQL/test/tRecordGram.cc:199:6: warning: ‘%s’ directive writing 218 bytes into a region of size 191 [-Wformat-overflow=] 
 void testExpr2()                                                                                                                                        
      ^~~~~~~~~                                                                                                                                          
In file included from /usr/include/stdio.h:862:0,                                                                                                        
                 from /usr/include/c++/7/cstdio:42,                                                                                                      
                 from /usr/include/c++/7/ext/string_conversions.h:43,                                                                                    
                 from /usr/include/c++/7/bits/basic_string.h:6352,                                                                                       
                 from /usr/include/c++/7/string:52,                                                                                                      
                 from /home/gijs/Work/casacore/casacore/casa/BasicSL/String.h:35,                                                                        
                 from /home/gijs/Work/casacore/casacore/casa/Containers/RecordInterface.h:36,                                                            
                 from /home/gijs/Work/casacore/casacore/tables/Tables/TableRecord.h:35,                                                                  
                 from /home/gijs/Work/casacore/tables/TaQL/test/tRecordGram.cc:28:                                                                       
/usr/include/x86_64-linux-gnu/bits/stdio2.h:34:43: note: ‘__builtin___sprintf_chk’ output 284 bytes into a destination of size 256                       
       __bos (__s), __fmt, __va_arg_pack ());                                                                                                            
                                           ^                                                                                                             
```

```
/home/gijs/Work/casacore/scimath_f/dfftpak.f:82:72:                                                 
                                                                                                    
       CALL DRFTI1 (N,WSAVE(N+1),WSAVE(2*N+1))                                                      
                                                                        1                           
Warning: Type mismatch in argument ‘ifac’ at (1); passed REAL(8) to INTEGER(4) [-Wargument-mismatch]
/home/gijs/Work/casacore/scimath_f/dfftpak.f:214:72:                                                
                                                                                                    
       CALL DRFTF1 (N,R,WSAVE,WSAVE(N+1),WSAVE(2*N+1))                                              
                                                                        1                           
Warning: Type mismatch in argument ‘ifac’ at (1); passed REAL(8) to INTEGER(4) [-Wargument-mismatch]
/home/gijs/Work/casacore/scimath_f/dfftpak.f:732:72:                                                
                                                                                                    
       CALL DRFTB1 (N,R,WSAVE,WSAVE(N+1),WSAVE(2*N+1))                                              
                                                                        1                           
Warning: Type mismatch in argument ‘ifac’ at (1); passed REAL(8) to INTEGER(4) [-Wargument-mismatch]
/home/gijs/Work/casacore/scimath_f/dfftpak.f:1284:72:                                               
                                                                                                    
       CALL DEFFT1 (N,WSAVE(2*N+1),WSAVE(3*N+1))                                                    
                                                                        1                           
Warning: Type mismatch in argument ‘ifac’ at (1); passed REAL(8) to INTEGER(4) [-Wargument-mismatch]
/home/gijs/Work/casacore/scimath_f/dfftpak.f:2213:72:                                               
                                                                                                    
       CALL DCFTI1 (N,WSAVE(IW1),WSAVE(IW2))                                                        
                                                                        1                           
Warning: Type mismatch in argument ‘ifac’ at (1); passed REAL(8) to INTEGER(4) [-Wargument-mismatch]
/home/gijs/Work/casacore/scimath_f/dfftpak.f:2341:72:                                               
                                                                                                    
       CALL DCFTF1 (N,C,WSAVE,WSAVE(IW1),WSAVE(IW2))                                                
                                                                        1                           
Warning: Type mismatch in argument ‘ifac’ at (1); passed REAL(8) to INTEGER(4) [-Wargument-mismatch]
/home/gijs/Work/casacore/scimath_f/dfftpak.f:2806:72:                                               
                                                                                                    
       CALL DCFTB1 (N,C,WSAVE,WSAVE(IW1),WSAVE(IW2))                                                
                                                                        1                           
Warning: Type mismatch in argument ‘ifac’ at (1); passed REAL(8) to INTEGER(4) [-Wargument-mismatch]

```


```
/home/gijs/Work/casacore/scimath_f/lawson.f:358:44:                                                     
                                                                                                        
       CALL H12 (1,NPP1,NPP1+1,M,A(1,J),1,UP,DUMMY,1,1,0)                                               
                                            1                                                           
Warning: Rank mismatch in argument ‘c’ at (1) (rank-1 and scalar) [-Wargument-mismatch]                 
/home/gijs/Work/casacore/scimath_f/lawson.f:886:45:                                                     
                                                                                                        
       CALL DH12 (1,NPP1,NPP1+1,M,A(1,J),1,UP,DUMMY,1,1,0)                                              
                                             1                                                          
Warning: Rank mismatch in argument ‘c’ at (1) (rank-1 and scalar) [-Wargument-mismatch]                 
[ 52%] Built target tRecordGram                                                                         
/home/gijs/Work/casacore/scimath_f/fftpak.f:92:72:                                                      
                                                                                                        
       CALL RFFTI1_OLD (N,WSAVE(N+1),WSAVE(2*N+1))                                                      
                                                                        1                               
Warning: Type mismatch in argument ‘ifac’ at (1); passed REAL(4) to INTEGER(4) [-Wargument-mismatch]    
/home/gijs/Work/casacore/scimath_f/fftpak.f:218:72:                                                     
                                                                                                        
       CALL RFFTF1_OLD (N,R,WSAVE,WSAVE(N+1),WSAVE(2*N+1))                                              
                                                                        1                               
Warning: Type mismatch in argument ‘ifac’ at (1); passed REAL(4) to INTEGER(4) [-Wargument-mismatch]    
/home/gijs/Work/casacore/scimath_f/fftpak.f:692:72:                                                     
                                                                                                        
       CALL RFFTB1_OLD (N,R,WSAVE,WSAVE(N+1),WSAVE(2*N+1))                                              
                                                                        1                               
Warning: Type mismatch in argument ‘ifac’ at (1); passed REAL(4) to INTEGER(4) [-Wargument-mismatch]    
/home/gijs/Work/casacore/scimath_f/fftpak.f:1134:72:                                                    
                                                                                                        
       CALL EZFFT1 (N,WSAVE(2*N+1),WSAVE(3*N+1))                                                        
                                                                        1                               
Warning: Type mismatch in argument ‘ifac’ at (1); passed REAL(4) to INTEGER(4) [-Wargument-mismatch]    
/home/gijs/Work/casacore/scimath_f/fftpak.f:1479:72:                                                    
                                                                                                        
       CALL SINT1(N,X,WSAVE,WSAVE(IW1),WSAVE(IW2),WSAVE(IW3))                                           
                                                                        1                               

```

```
Warning: Type mismatch in argument ‘c’ at (1); passed REAL(4) to COMPLEX(4) [-Wargument-mismatch]   
/home/gijs/Work/casacore/scimath_f/fftpak.f:10379:50:                                               
                                                                                                    
       CALL RFFT1F(NP1,1,XH,LNXH,WSAVE(NS2+1),LNSV,WORK,LNWK,IER1)                                  
                                                  1                                                 
Warning: Rank mismatch in argument ‘work’ at (1) (rank-1 and scalar) [-Wargument-mismatch]          
/home/gijs/Work/casacore/scimath_f/fftpak.f:10437:50:                                               
                                                                                                    
       CALL RFFT1F(NP1,1,XH,LNXH,WSAVE(NS2+1),LNSV,WORK,                                            
                                                  1                                                 
Warning: Rank mismatch in argument ‘work’ at (1) (rank-1 and scalar) [-Wargument-mismatch]          
/home/gijs/Work/casacore/scimath_f/fftpak.f:10495:72:                                               
                                                                                                    
       CALL MSNTB1(LOT,JUMP,N,INC,X,WSAVE,WORK,WORK(IW1),WORK(IW2),IER1)                            
                                                                        1                           
Warning: Type mismatch in argument ‘dsum’ at (1); passed REAL(4) to REAL(8) [-Wargument-mismatch]   
/home/gijs/Work/casacore/scimath_f/fftpak.f:10540:72:                                               
                                                                                                    
       CALL MSNTF1(LOT,JUMP,N,INC,X,WSAVE,WORK,WORK(IW1),WORK(IW2),IER1)                            
                                                                        1                           
Warning: Type mismatch in argument ‘dsum’ at (1); passed REAL(4) to REAL(8) [-Wargument-mismatch]   
```

```
/home/gijs/Work/casacore/ms/MSOper/test/tNewMSSimulator.cc:105:8: warning: ‘template<class> class std::auto_ptr’ is deprecated [-Wdeprecated-declarations]     
   std::auto_ptr<casacore::NewMSSimulator> simulator_p;                                                                                                        
        ^~~~~~~~                                                                                                                                               
In file included from /usr/include/c++/7/memory:80:0,                                                                                                          
                 from /home/gijs/Work/casacore/casacore/ms/MSOper/NewMSSimulator.h:33,                                                                         
                 from /home/gijs/Work/casacore/ms/MSOper/test/tNewMSSimulator.cc:34:                                                                           
/usr/include/c++/7/bits/unique_ptr.h:51:28: note: declared here                                                                                                
   template<typename> class auto_ptr;                                                                                                                          
                            ^~~~~~~~                                                                                                                           
```

Compilers are getting pickier all the time. I'll fix the %i issue.
The mirlib warnings are around for a long time; this is imported ancient
code. I'll see if I can fix then some time, but not now.
There are also warnings coming from the code generated by bison; nothing we
can do about it.

On Wed, Oct 31, 2018 at 12:21 PM Gijs Molenaar <notifications@github.com>
wrote:

> If i remember correctly casacore has always been quite tidy with compile
> time warnings, but if I compile the latest master I encounter a lot of
> warnings. We should probably solve these before the 2.5 releasae.
>
> /home/gijs/Work/casacore/casa/Containers/RecordDescRep.cc: In member function ‘casacore::String casacore::RecordDescRep::uniqueName(const casacore::String&) const’:
> /home/gijs/Work/casacore/casa/Containers/RecordDescRep.cc:374:8: warning: ‘%i’ directive writing between 1 and 10 bytes into a region of size 7 [-Wformat-overflow=]
>  String RecordDescRep::uniqueName (const String& name) const
>         ^~~~~~~~~~~~~
> /home/gijs/Work/casacore/casa/Containers/RecordDescRep.cc:374:8: note: directive argument in the range [1, 2147483647]
> In file included from /usr/include/stdio.h:862:0,
>                  from /usr/include/c++/7/cstdio:42,
>                  from /usr/include/c++/7/ext/string_conversions.h:43,
>                  from /usr/include/c++/7/bits/basic_string.h:6352,
>                  from /usr/include/c++/7/string:52,
>                  from /usr/include/c++/7/bits/locale_classes.h:40,
>                  from /usr/include/c++/7/bits/ios_base.h:41,
>                  from /usr/include/c++/7/ios:42,
>                  from /usr/include/c++/7/istream:38,
>                  from /usr/include/c++/7/sstream:38,
>                  from /usr/include/c++/7/complex:45,
>                  from /home/gijs/Work/casacore/casacore/casa/BasicSL/Complexfwd.h:45,
>                  from /home/gijs/Work/casacore/casacore/casa/BasicSL/Complex.h:35,
>                  from /home/gijs/Work/casacore/casacore/casa/Utilities/DataType.h:32,
>                  from /home/gijs/Work/casacore/casacore/casa/Containers/RecordDescRep.h:35,
>                  from /home/gijs/Work/casacore/casa/Containers/RecordDescRep.cc:28:
> /usr/include/x86_64-linux-gnu/bits/stdio2.h:34:43: note: ‘__builtin___sprintf_chk’ output between 3 and 12 bytes into a destination of size 8
>        __bos (__s), __fmt, __va_arg_pack ());
>                                            ^
>
> In member function ‘virtual void casacore::MultiFile::extend(casacore::MultiFileInfo&, casacore::Int64)’:
> cc1plus: warning: ‘void* __builtin_memset(void*, int, long unsigned int)’: specified size 18446744073709551608 exceeds maximum object size 9223372036854775807 [-Wstringop-overflow=]
>
> /home/gijs/Work/casacore/mirlib/key.c:439:9: warning: ignoring return value of ‘system’, declared with attribute warn_unused_result [-Wunused-result]
>          (void)system(string);
>          ^~~~~~~~~~~~~~~~~~~~
> /home/gijs/Work/casacore/mirlib/key.c:443:9: warning: ignoring return value of ‘system’, declared with attribute warn_unused_result [-Wunused-result]
>          (void)system(string);
>          ^~~~~~~~~~~~~~~~~~~~
>
> /home/gijs/Work/casacore/mirlib/uvio.c: In function ‘uvopen_c’:
> /home/gijs/Work/casacore/mirlib/uvio.c:749:29: warning: passing argument 3 of ‘rdhdl_c’ from incompatible pointer type [-Wincompatible-pointer-types]
>        rdhdl_c(*tno,"vislen",&(uv->max_offset),hsize_c(uv->item));
>                              ^
> In file included from /home/gijs/Work/casacore/mirlib/uvio.c:288:0:
> /home/gijs/Work/casacore/mirlib/miriad.h:182:6: note: expected ‘int8 * {aka long long int *}’ but argument is of type ‘off_t * {aka long int *}’
>  void rdhdl_c (int tno, Const char *keyword, int8 *value, int8 defval);
>       ^~~~~~~
> /home/gijs/Work/casacore/mirlib/uvio.c:794:29: warning: passing argument 3 of ‘rdhdl_c’ from incompatible pointer type [-Wincompatible-pointer-types]
>        rdhdl_c(*tno,"vislen",&(uv->offset),hsize_c(uv->item));
>                              ^
> In file included from /home/gijs/Work/casacore/mirlib/uvio.c:288:0:
> /home/gijs/Work/casacore/mirlib/miriad.h:182:6: note: expected ‘int8 * {aka long long int *}’ but argument is of type ‘off_t * {aka long int *}’
>  void rdhdl_c (int tno, Const char *keyword, int8 *value, int8 defval);
>       ^~~~~~~
> /home/gijs/Work/casacore/mirlib/uvio.c:814:26: warning: passing argument 3 of ‘rdhdl_c’ from incompatible pointer type [-Wincompatible-pointer-types]
>      rdhdl_c(*tno,"ncorr",&(uv->corr_flags.offset),-1);
>                           ^
> In file included from /home/gijs/Work/casacore/mirlib/uvio.c:288:0:
> /home/gijs/Work/casacore/mirlib/miriad.h:182:6: note: expected ‘int8 * {aka long long int *}’ but argument is of type ‘off_t * {aka long int *}’
>  void rdhdl_c (int tno, Const char *keyword, int8 *value, int8 defval);
>       ^~~~~~~
> /home/gijs/Work/casacore/mirlib/uvio.c:815:27: warning: passing argument 3 of ‘rdhdl_c’ from incompatible pointer type [-Wincompatible-pointer-types]
>      rdhdl_c(*tno,"nwcorr",&(uv->wcorr_flags.offset),-1);
>                            ^
> In file included from /home/gijs/Work/casacore/mirlib/uvio.c:288:0:
> /home/gijs/Work/casacore/mirlib/miriad.h:182:6: note: expected ‘int8 * {aka long long int *}’ but argument is of type ‘off_t * {aka long int *}’
>  void rdhdl_c (int tno, Const char *keyword, int8 *value, int8 defval);
>       ^~~~~~~
>
> /home/gijs/Work/casacore/tables/Tables/ReadAsciiTable.cc: In static member function ‘static void casacore::ReadAsciiTable::getTypes(const casacore::IPosition&, const Char*, casacore::Int, casacore::Char*, casacore::Char*│
> , casacore::Char)’:                                                                                                                                                                                                         │
> /home/gijs/Work/casacore/tables/Tables/ReadAsciiTable.cc:164:6: warning: ‘%i’ directive writing between 1 and 10 bytes into a region of size 9 [-Wformat-overflow=]                                                         ├
>  void ReadAsciiTable::getTypes (const IPosition& shape,                                                                                                                                                                     │
>       ^~~~~~~~~~~~~~                                                                                                                                                                                                        │
> /home/gijs/Work/casacore/tables/Tables/ReadAsciiTable.cc:164:6: note: directive argument in the range [1, 2147483647]
> In file included from /usr/include/stdio.h:862:0,                                                                                                                                                                           │
>                  from /usr/include/c++/7/cstdio:42,                                                                                                                                                                         │
>                  from /usr/include/c++/7/ext/string_conversions.h:43,                                                                                                                                                       │
>                  from /usr/include/c++/7/bits/basic_string.h:6352,                                                                                                                                                          │
>                  from /usr/include/c++/7/string:52,                                                                                                                                                                         │
>                  from /home/gijs/Work/casacore/casacore/casa/BasicSL/String.h:35,                                                                                                                                           │
>                  from /home/gijs/Work/casacore/casacore/tables/Tables/ReadAsciiTable.h:33,                                                                                                                                  │
>                  from /home/gijs/Work/casacore/tables/Tables/ReadAsciiTable.cc:28:
> /usr/include/x86_64-linux-gnu/bits/stdio2.h:34:43: note: ‘__builtin___sprintf_chk’ output between 9 and 18 bytes into a destination of size 16                                                                              │
>        __bos (__s), __fmt, __va_arg_pack ());
>                                            ^
>
> /home/gijs/Work/casacore/tables/TaQL/test/tRecordGram.cc: In function ‘void testExpr2()’:
> /home/gijs/Work/casacore/tables/TaQL/test/tRecordGram.cc:199:6: warning: ‘%s’ directive writing 218 bytes into a region of size 191 [-Wformat-overflow=]
>  void testExpr2()
>       ^~~~~~~~~
> In file included from /usr/include/stdio.h:862:0,
>                  from /usr/include/c++/7/cstdio:42,
>                  from /usr/include/c++/7/ext/string_conversions.h:43,
>                  from /usr/include/c++/7/bits/basic_string.h:6352,
>                  from /usr/include/c++/7/string:52,
>                  from /home/gijs/Work/casacore/casacore/casa/BasicSL/String.h:35,
>                  from /home/gijs/Work/casacore/casacore/casa/Containers/RecordInterface.h:36,
>                  from /home/gijs/Work/casacore/casacore/tables/Tables/TableRecord.h:35,
>                  from /home/gijs/Work/casacore/tables/TaQL/test/tRecordGram.cc:28:
> /usr/include/x86_64-linux-gnu/bits/stdio2.h:34:43: note: ‘__builtin___sprintf_chk’ output 284 bytes into a destination of size 256
>        __bos (__s), __fmt, __va_arg_pack ());
>                                            ^
>
> /home/gijs/Work/casacore/scimath_f/dfftpak.f:82:72:
>
>        CALL DRFTI1 (N,WSAVE(N+1),WSAVE(2*N+1))
>                                                                         1
> Warning: Type mismatch in argument ‘ifac’ at (1); passed REAL(8) to INTEGER(4) [-Wargument-mismatch]
> /home/gijs/Work/casacore/scimath_f/dfftpak.f:214:72:
>
>        CALL DRFTF1 (N,R,WSAVE,WSAVE(N+1),WSAVE(2*N+1))
>                                                                         1
> Warning: Type mismatch in argument ‘ifac’ at (1); passed REAL(8) to INTEGER(4) [-Wargument-mismatch]
> /home/gijs/Work/casacore/scimath_f/dfftpak.f:732:72:
>
>        CALL DRFTB1 (N,R,WSAVE,WSAVE(N+1),WSAVE(2*N+1))
>                                                                         1
> Warning: Type mismatch in argument ‘ifac’ at (1); passed REAL(8) to INTEGER(4) [-Wargument-mismatch]
> /home/gijs/Work/casacore/scimath_f/dfftpak.f:1284:72:
>
>        CALL DEFFT1 (N,WSAVE(2*N+1),WSAVE(3*N+1))
>                                                                         1
> Warning: Type mismatch in argument ‘ifac’ at (1); passed REAL(8) to INTEGER(4) [-Wargument-mismatch]
> /home/gijs/Work/casacore/scimath_f/dfftpak.f:2213:72:
>
>        CALL DCFTI1 (N,WSAVE(IW1),WSAVE(IW2))
>                                                                         1
> Warning: Type mismatch in argument ‘ifac’ at (1); passed REAL(8) to INTEGER(4) [-Wargument-mismatch]
> /home/gijs/Work/casacore/scimath_f/dfftpak.f:2341:72:
>
>        CALL DCFTF1 (N,C,WSAVE,WSAVE(IW1),WSAVE(IW2))
>                                                                         1
> Warning: Type mismatch in argument ‘ifac’ at (1); passed REAL(8) to INTEGER(4) [-Wargument-mismatch]
> /home/gijs/Work/casacore/scimath_f/dfftpak.f:2806:72:
>
>        CALL DCFTB1 (N,C,WSAVE,WSAVE(IW1),WSAVE(IW2))
>                                                                         1
> Warning: Type mismatch in argument ‘ifac’ at (1); passed REAL(8) to INTEGER(4) [-Wargument-mismatch]
>
>
> /home/gijs/Work/casacore/scimath_f/lawson.f:358:44:
>
>        CALL H12 (1,NPP1,NPP1+1,M,A(1,J),1,UP,DUMMY,1,1,0)
>                                             1
> Warning: Rank mismatch in argument ‘c’ at (1) (rank-1 and scalar) [-Wargument-mismatch]
> /home/gijs/Work/casacore/scimath_f/lawson.f:886:45:
>
>        CALL DH12 (1,NPP1,NPP1+1,M,A(1,J),1,UP,DUMMY,1,1,0)
>                                              1
> Warning: Rank mismatch in argument ‘c’ at (1) (rank-1 and scalar) [-Wargument-mismatch]
> [ 52%] Built target tRecordGram
> /home/gijs/Work/casacore/scimath_f/fftpak.f:92:72:
>
>        CALL RFFTI1_OLD (N,WSAVE(N+1),WSAVE(2*N+1))
>                                                                         1
> Warning: Type mismatch in argument ‘ifac’ at (1); passed REAL(4) to INTEGER(4) [-Wargument-mismatch]
> /home/gijs/Work/casacore/scimath_f/fftpak.f:218:72:
>
>        CALL RFFTF1_OLD (N,R,WSAVE,WSAVE(N+1),WSAVE(2*N+1))
>                                                                         1
> Warning: Type mismatch in argument ‘ifac’ at (1); passed REAL(4) to INTEGER(4) [-Wargument-mismatch]
> /home/gijs/Work/casacore/scimath_f/fftpak.f:692:72:
>
>        CALL RFFTB1_OLD (N,R,WSAVE,WSAVE(N+1),WSAVE(2*N+1))
>                                                                         1
> Warning: Type mismatch in argument ‘ifac’ at (1); passed REAL(4) to INTEGER(4) [-Wargument-mismatch]
> /home/gijs/Work/casacore/scimath_f/fftpak.f:1134:72:
>
>        CALL EZFFT1 (N,WSAVE(2*N+1),WSAVE(3*N+1))
>                                                                         1
> Warning: Type mismatch in argument ‘ifac’ at (1); passed REAL(4) to INTEGER(4) [-Wargument-mismatch]
> /home/gijs/Work/casacore/scimath_f/fftpak.f:1479:72:
>
>        CALL SINT1(N,X,WSAVE,WSAVE(IW1),WSAVE(IW2),WSAVE(IW3))
>                                                                         1
>
>
> Warning: Type mismatch in argument ‘c’ at (1); passed REAL(4) to COMPLEX(4) [-Wargument-mismatch]
> /home/gijs/Work/casacore/scimath_f/fftpak.f:10379:50:
>
>        CALL RFFT1F(NP1,1,XH,LNXH,WSAVE(NS2+1),LNSV,WORK,LNWK,IER1)
>                                                   1
> Warning: Rank mismatch in argument ‘work’ at (1) (rank-1 and scalar) [-Wargument-mismatch]
> /home/gijs/Work/casacore/scimath_f/fftpak.f:10437:50:
>
>        CALL RFFT1F(NP1,1,XH,LNXH,WSAVE(NS2+1),LNSV,WORK,
>                                                   1
> Warning: Rank mismatch in argument ‘work’ at (1) (rank-1 and scalar) [-Wargument-mismatch]
> /home/gijs/Work/casacore/scimath_f/fftpak.f:10495:72:
>
>        CALL MSNTB1(LOT,JUMP,N,INC,X,WSAVE,WORK,WORK(IW1),WORK(IW2),IER1)
>                                                                         1
> Warning: Type mismatch in argument ‘dsum’ at (1); passed REAL(4) to REAL(8) [-Wargument-mismatch]
> /home/gijs/Work/casacore/scimath_f/fftpak.f:10540:72:
>
>        CALL MSNTF1(LOT,JUMP,N,INC,X,WSAVE,WORK,WORK(IW1),WORK(IW2),IER1)
>                                                                         1
> Warning: Type mismatch in argument ‘dsum’ at (1); passed REAL(4) to REAL(8) [-Wargument-mismatch]
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/casacore/casacore/issues/794>, or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AHfIPN6RCKvEnhePomHzfeuF4iRtn93eks5uqYe8gaJpZM4YD3Kq>
> .
>

--- a/casa/Containers/RecordDescRep.cc
+++ b/casa/Containers/RecordDescRep.cc
@@ -374,7 +374,7 @@ String RecordDescRep::makeName (Int whichField) const
 String RecordDescRep::uniqueName (const String& name) const
 {
     String newName = name;
-    char strc[8];      // should be plenty large...
+    char strc[16];      // should be plenty large...
     int n = 0;
     while (fieldNumber(newName) >= 0) {
 	++n;

--- a/tables/TaQL/TaQLShow.cc
+++ b/tables/TaQL/TaQLShow.cc
@@ -1037,7 +1037,7 @@ namespace casacore { //# NAMESPACE CASACORE - BEGIN
   };
 
   const char* radialVelocityHelp[] = {
-    "'RadialVelocity types",
+    "RadialVelocity types",
     "    LSRD     Local Standard of Rest (J2000) - dynamical definition (IAU, [9,12,7] km/s in galactic coordinates)",
     "    LSRK     LSR as kinematical (radio) definition - 20.0 km/s in direction ra,dec = [270,+30] deg (B1900.0)",
     "    BARY     Barycentric (J2000)",

--- a/tables/Tables/ReadAsciiTable.cc
+++ b/tables/Tables/ReadAsciiTable.cc
@@ -185,7 +185,7 @@ void ReadAsciiTable::getTypes (const IPosition& shape,
 	    }
 	}
 	string1++;
-	char name[16];
+	char name[24];
 	i++;
 	sprintf (name, " Column%i", i);
 	strcpy (string2, name);

